services/app/src/actions/Search.tsx

```diff
@@ -67,43 +68,35 @@ function getSearchResults(params: SearchParams, callback: (response: QuestSearch
     // Clear previous results
     dispatch({type: 'SEARCH_REQUEST'});
 
-    const xhr = new XMLHttpRequest();
-    // TODO: Pagination / infinite scrolling
-    xhr.open('POST', AUTH_SETTINGS.URL_BASE + '/quests', true);
-    xhr.setRequestHeader('Content-Type', 'text/plain');
-    xhr.onload = () => {
-      let response: QuestSearchResponse;
-      try {
-        response = JSON.parse(xhr.responseText);
-      } catch (e) {
-        response = {error: e.toString(), hasMore: false, quests: []};
-      }
-
-      // Log silently to console, so we can see if users file feedback
-      // with recurrent search problems.
-      if (response.error) {
-        console.error(response.error);
-      }
-
+    fetch(AUTH_SETTINGS.URL_BASE + '/quests', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'text/plain',
+      },
+      body: JSON.stringify(params),
+    })
+    .then(handleFetchErrors)
+    .then((response: Response) => response.json())
+    .then((data: QuestSearchResponse) => {
       // Simple validation of response quests
       const quests: Quest[] = [];
-      if (response.quests) {
-        for (const q of response.quests) {
+
+      if (data.quests) {
+        for (const q of data.quests) {
           const i = Quest.create(q);
           if (i instanceof Error) {
             console.error('Parsed invalid quest: ' + JSON.stringify(q));
-          } else {
-            quests.push(i);
+           } else {
```

Comment:
looks like some extra whitespace accidentally got inserted here?

Reply:
Yes, will take care of it.

Reply:
Hmm, still seeing the extra whitespace here?

Reply:
Now that we have prettier set up, can you run it through this file again to sort out the extraneous spaces?